network/struct.go

```diff
@@ -149,6 +179,42 @@ func (si *ServerIdentity) GetPrivate() kyber.Scalar {
 	return si.private
 }
 
+// ServicePublic returns the public key of the service or the default
+// one if the service has not been registered with a suite
+func (si *ServerIdentity) ServicePublic(name string) kyber.Point {
+	for _, si := range si.ServiceIdentities {
```

Comment:
Please don't shadow variables quite so close to each other. Suggestion: use svcid for the iterator here.